muacrypt/recommendation.py

```diff
@@ -1,34 +1,66 @@
 class Recommendation:
     """ Calculating recommendations for encryption """
 
-    def __init__(self, peerstates):
+    def __init__(self, peerstates, prefer_encrypt, reply_to_enc=False):
         self.peerstates = peerstates
+        self.prefer_encrypt = prefer_encrypt
+        self.reply_to_enc = reply_to_enc
 
     def ui_recommendation(self):
         # only consider first peer for now
         peer = list(self.peerstates.values())[0]
-        return self._peer_recommendation(peer)
+        return self._peer_recommendation(peer).ui_recommendation()
 
     def target_keys(self):
-        return {addr: self._target_key(peer) for addr, peer in
+        return {addr: self._peer_recommendation(peer).target_key()
+                for addr, peer in
                 self.peerstates.items()}
 
     def _peer_recommendation(self, peer):
-        if self._target_key(peer):
-            return 'available'
-        else:
+        return PeerRecommendation(peer, self.prefer_encrypt,
+                self.reply_to_enc)
+
+
+class PeerRecommendation:
+    """ Calculating recommendation for a single peer """
+
+    def __init__(self, peer, prefer_encrypt, reply_to_enc):
+        self.peer = peer
+        self.prefer_encrypt = prefer_encrypt
+        self.reply_to_enc = reply_to_enc
+
+    def ui_recommendation(self):
+        pre = self._preliminary_recommendation()
+        if ((pre == 'available' or pre == 'discourage') and
+                self.reply_to_enc):
+            return 'encrypt'
+        if (pre == 'available' and
+                self.prefer_encrypt == 'mutual' and
+                self.peer.prefer_encrypt == 'mutual'):
+            return 'encrypt'
+        return pre
+
+    def target_key(self):
+        return self._public_key() or self._gossip_key()
+
+    def _preliminary_recommendation(self):
+        if self.target_key() is None:
             return 'disable'
+        if self._ac_is_outdated():
+            return 'discourage'
+        return 'available'
 
-    def _target_key(self, peer):
-        return self._public_key(peer) or self._gossip_key(peer)
+    def _ac_is_outdated(self):
+        timeout = 35 * 24 * 60 * 60
+        return (self.peer.last_seen - self.peer.autocrypt_timestamp > timeout)
```

Comment:
i think this ``_ac_is_outdated`` should just be directly inside the preliminary_recommendation function.  having lots of little functions requires a lot of function ntesting reading.

Reply:
wrt to my "abstraction" gut feeling i mentioned on IRC: e.g.  introducing a super-small helper function separates concerns and makes the ``preliminary_recommendation`` more abstract than it needs to be at this point.   If it turns out we want to do more checks wrt "ac_is_outdated" we can always create that extra function then, i.e. "abstract/separate" on demand rather than ahead.  Also it's not clear what "ac_is_outdated" will do so one has to read that function -- and is it going to check also for gossip outdatedness?